bouncer/views.py

```diff
@@ -25,11 +29,20 @@ def annotation(self):
                 id=self.request.matchdict["id"])
         except exceptions.NotFoundError:
             statsd.incr("views.annotation.404.annotation_not_found")
-            raise httpexceptions.HTTPNotFound()
+            raise httpexceptions.HTTPNotFound(_("Annotation not found"))
 
         annotation_id = document["_id"]
         document_uri = document["_source"]["uri"]
 
+        if not (document_uri.startswith("http://") or
+                document_uri.startswith("https://")):
+            statsd.incr("views.annotation.422.not_an_http_or_https_document")
+            raise httpexceptions.HTTPUnprocessableEntity(
+                _("Sorry, but it looks like this annotation was made on a "
+                  "document that is not publicly available. "
+                  "To view it’s annotations, a document's address must start "
```

Comment:
Ugh, there's a typo and a non-ASCII character in this (thanks PyLint). My fault, will fix it